neo4j/resultsummary.go

```diff
@@ -179,6 +182,14 @@ type resultSummary struct {
 	params map[string]interface{}
 }
 
+func (s *resultSummary) Agent() string {
+	return s.sum.Agent
+}
+
+func (s *resultSummary) ProtocolVersion() string {
```

Comment:
Do we need to return a string? Could be 
ProtocolVersion() (major int, minor int)
Which would be cheaper internally and reduce need for parsing on consuming end